### PR TITLE
[SYCL] Move SYCL headers from CL/ to sycl/

### PR DESCRIPTION
**Pull request by aelovikov-intel**
`CL/sycl.hpp` is changed to include `sycl/sycl.hpp`. 

No deprecation warning emitted as the header is required by the SYCL 2020
for backward compatibility.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1078

**Comment by aelovikov-intel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1078

**Comment by aelovikov-intel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1078

**Comment by aelovikov-intel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1078

**Comment by aelovikov-intel** [comment]
This change becomes obsolete due to conflicts really fast, so I'd appreciate prompt review from all the reviewers.

Also, I think it might be better to get it merged in without striving for perfection after just a couple of review rounds and cleanup later as keeping it up-to-date would be really time consuming otherwise.

File: sycl/gdb/libsycl.so-gdb.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def index(self, arg):
             arg = arg["MImpl"]["MIndex"]
         except:
             pass
-        # https://github.com/intel/llvm/blob/97272b7ebd569bfa13811913a31e30f926559217/sycl/include/CL/sycl/accessor.hpp#L678-L690
+        # https://github.com/intel/llvm/blob/97272b7ebd569bfa13811913a31e30f926559217/sycl/include/sycl/accessor.hpp#L678-L690
```
Comment (v-klochkov) [inline sycl/gdb/libsycl.so-gdb.py:38]: The link here and below is 404. Please update or remove.

**Review by v-klochkov — APPROVED** [review]
ESIMD: looks good.

File: sycl/gdb/libsycl.so-gdb.py (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/gdb/libsycl.so-gdb.py:38]: I'll remove once pre-commit testing finishes, thanks!

**Comment by asudarsa** [comment]
Hi @aelovikov-intel 
Thanks for putting this up. One quick question. In addition to '[CL/sycl/* -> sycl/]', I think you are also moving CL/* (not including sycl dir) into sycl/. Is my understanding correct?
Thanks

**Comment by aelovikov-intel** [comment]
> Hi @aelovikov-intel Thanks for putting this up. One quick question. In addition to '[CL/sycl/* -> sycl/]', I think you are also moving CL/* (not including sycl dir) into sycl/. Is my understanding correct? Thanks

I'm not sure what you mean. `CL/` contains three entries (https://github.com/intel/llvm/tree/sycl/sycl/include/CL):
* `__spirv/` - not touched
* `sycl/` - the one being moved in this PR, i.e. all files (`CL/sycl/*`) are in `llvm/sycl/include/sycl/*` now
* `sycl.hpp` - content moved to `llvm/sycl/include/sycl/sycl.hpp` and replaced by single `#include <sycl/sycl.hpp>`

**Review by asudarsa — DISMISSED** [review]
LGTM. Thanks

**Comment by aelovikov-intel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1078

**Review by cperkinsintel — DISMISSED** [review]
LGTM

**Review by premanandrao — APPROVED** [review]
FE test changes look okay to me.

**Review by smaslov-intel — DISMISSED** [review]

**Review by smanna12 — DISMISSED** [review]
FE changes LGTM

**Review by bader — DISMISSED** [review]

**Review by bader — COMMENTED** [review]
Revert of llvm-spirv changes looks good to me. Let's consider that all approves are still in place.
If @intel/dpcpp-specification-reviewers team approves, we can merge this PR.

**Comment by aelovikov-intel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1078

**Comment by aelovikov-intel** [comment]
@intel/dpcpp-specification-reviewers , can you please take a look at the relevant part of this?

**Review by asudarsa — APPROVED** [review]
llvm-SPIRV changes LGTM. Thanks

**Review by v-klochkov — DISMISSED** [review]

**Review by smanna12 — DISMISSED** [review]

**Review by jbrodman — DISMISSED** [review]
LGTM

**Review by smanna12 — APPROVED** [review]

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**